test/runner/runner.go

```diff
@@ -26,22 +26,58 @@ import (
 func NewRegistry() *TestRegistry {
 	return &TestRegistry{
 		tests: make(map[string]Test),
+		testGroups: make(map[string]TestGroup),
 	}
 }
 
-// TestRegistry contains a mapping of named tests
+//NewTestGroup returns a pointer to a new TestGroup
+func NewTestGroup(name string) *TestGroup {
+	return &TestGroup{
+		name: name,
+		tests: make(map[string]Test),
+	}
+}
+
+// Test is a test function
+type Test func(t *testing.T)
+
+
+// TestRegistry contains a mapping of named test groups
 type TestRegistry struct {
 	suite.SetupTestSuite
 	tests map[string]Test
+	testGroups map[string]TestGroup
 }
 
-// Register registers a named test
+//TestGroup to run multiple tests
+type TestGroup struct {
+	name string
+	suite.SetupTestSuite
+	tests map[string]Test
+}
+
+// AddToGroup registers a named test
+func (r *TestRegistry) AddToGroup(testName string, testGroup *TestGroup, test Test) {
```

Comment:
This seems like it would make more sense as a receiver on `TestGroup` rather than `TestRegistry` since it's actually modifying the state of the group.